src/sign.rs

```diff
@@ -71,25 +87,25 @@ impl<'a> TransactionData<'a> {
         s.append(&self.to);
         s.append(&self.value);
         s.append(&self.data.0);
-        s.append(&v);
-        s.append(&U256::from(sig.r));
-        s.append(&U256::from(sig.s));
+        s.append(&sig_v);
+        s.append(&U256::from(sig_r));
+        s.append(&U256::from(sig_s));
     }
 }
 
 /// Encode chain ID based on (EIP-155)[https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md)
-fn add_chain_replay_protection(v: u64, chain_id: Option<u64>) -> u64 {
-    v + if let Some(n) = chain_id {
-        35 + n * 2
-    } else {
-        27
-    }
+fn add_chain_replay_protection(recovery_id: RecoveryId, chain_id: Option<u64>) -> u64 {
+    (recovery_id.to_i32() as u64)
+        + if let Some(n) = chain_id {
+            35 + n * 2
+        } else {
+            27
+        }
```

Comment:
awkward... I guess this is EIP-155 specific.

Reply:
Yes. It's for chain replay protection - its so you can't use the same transaction on multiple chains. For example if you sign a transaction on rinkeby, you wouldn't expect the same transaction to be usable on mainnet. This used to be possible before EIP-155 since when using the same private keys for both networks (since the signature would be the same for the same nonce). This makes it so the chain ID is part of the signature to prevent that from happening.